commit 98db090a73133d72e4f22943908f31eb0298e120
Author: Nathan Chancellor <natechancellor@gmail.com>
Date:   Thu Sep 27 17:50:39 2018 -0700

    staging: bcm2835-camera: Avoid unneeded internal declaration warning
    
    Clang warns:
    
    drivers/staging/vc04_services/bcm2835-camera/controls.c:59:18: warning:
    variable 'mains_freq_qmenu' is not needed and will not be emitted
    [-Wunneeded-internal-declaration]
    static const s64 mains_freq_qmenu[] = {
                     ^
    1 warning generated.
    
    This is because mains_freq_qmenu is currently only used in an ARRAY_SIZE
    macro, which is a compile time evaluation in this case. Avoid this by
    adding mains_freq_qmenu as the imenu member of this structure, which
    matches all other controls that uses the ARRAY_SIZE macro in v4l2_ctrls.
    This turns out to be a no-op because V4L2_CID_MPEG_VIDEO_BITRATE_MODE is
    defined as a MMAL_CONTROL_TYPE_STD_MENU, which does not pass the imenu
    definition along to v4l2_ctrl_new in bm2835_mmal_init_controls.
    
    Link: https://github.com/ClangBuiltLinux/linux/issues/122
    Signed-off-by: Nathan Chancellor <natechancellor@gmail.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/drivers/staging/vc04_services/bcm2835-camera/controls.c b/drivers/staging/vc04_services/bcm2835-camera/controls.c
index 8f23a7d3119c..fb86e92faffa 100644
--- a/drivers/staging/vc04_services/bcm2835-camera/controls.c
+++ b/drivers/staging/vc04_services/bcm2835-camera/controls.c
@@ -1109,7 +1109,7 @@ static const struct bm2835_mmal_v4l2_ctrl v4l2_ctrls[V4L2_CTRL_COUNT] = {
 	{
 		V4L2_CID_POWER_LINE_FREQUENCY, MMAL_CONTROL_TYPE_STD_MENU,
 		0, ARRAY_SIZE(mains_freq_qmenu) - 1,
-		1, 1, NULL,
+		1, 1, mains_freq_qmenu,
 		MMAL_PARAMETER_FLICKER_AVOID,
 		&ctrl_set_flicker_avoidance,
 		false
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1402_98db090a73133d72e4f22943908f31eb0298e120.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1402_98db090a73133d72e4f22943908f31eb0298e120.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
